### DISCUSSION TITLE
Suggestion to provide official docker image

### DISCUSSION THREAD

**Discussion by DarkPark** [google/zx#996] [General] [answered]
Node.js is not always available on systems, but sometimes Docker is installed instead.
Therefore, it is possible to use `zx` from a Docker container (if one is provided).
It's not difficult. I've tested it. Here is the recipe.

Dockerfile:
```
FROM node:lts-slim

RUN npm install -g zx

ENTRYPOINT ["zx"]
```

Build the image with:
 `docker build --tag zx .`

Run any script in the local folder (e.g., using the Fish shell):
```shell
docker run --interactive --tty --rm \
  --user (id -u "$USER"):(id -g "$USER") \
  --volume "$PWD":/tmp \
  --workdir /tmp \
  --network=host \
  zx test.js
```

Or add this command as shell alias:
`alias zx='docker run --interactive --tty --rm --user (id -u "$USER"):(id -g "$USER") --volume "$PWD":/tmp --workdir /tmp --network=host zx'`

now it's possible to run zx with a simple command:
```shell
zx ./test.js
```

I'm ready to publish such a Docker image, but seems to me it would be more appropriate if you provided and supported it.

**Comment by antongolub** [comment]
And that's pretty easy to maintain in CI/CD.

**Reply by DarkPark** [reply]
@antongolub indeed it is ) 
So will you be able to give it a go?

**Reply by antongolub** [reply]
It's time when I can say: yes
https://github.com/google/zx/pull/1142

**Reply by antongolub** [reply, marked as answer]
@DarkPark,

https://github.com/google/zx/releases/tag/8.5.0
https://github.com/google/zx/pkgs/container/zx

**Reply by DarkPark** [reply]
@antongolub nice! thank you )
